heareval/embeddings/task_embeddings.py

```diff
@@ -217,14 +219,97 @@ def get_labels_for_timestamps(labels: List, timestamps: np.ndarray) -> List:
         # Update the binary vector of labels with intervals for each timestamp
         for j, t in enumerate(timestamps[i]):
             interval_labels = [interval.data for interval in tree[t]]
-            labels_for_sound.append([float(t), interval_labels])
+            labels_for_sound.append(interval_labels)
+            # If we want to store the timestamp too
+            # labels_for_sound.append([float(t), interval_labels])
 
         timestamp_labels.append(labels_for_sound)
 
     return timestamp_labels
 
 
+def memmap_embeddings(
+    outdir: Path,
+    prng: random.Random,
+    metadata: Dict,
+    split_name: str,
+    embed_dir: Path,
+    task_name: str,
+):
+    """
+    Memmap all the embeddings to one file, and pickle all the labels.
+    (We assume labels can fit in memory.)
+    TODO: This writes things to disk double, we could clean that up after.
+    We might also be able to get away with writing to disk only once.
+    """
+    embedding_files = list(outdir.glob("*.embedding.npy"))
+    prng.shuffle(embedding_files)
+
+    # First count the number of embeddings total
+    nembeddings = 0
+    ndim = None
+    dtype = None
+    for embedding_file in tqdm(embedding_files):
+        emb = np.load(embedding_file)
+        if metadata["embedding_type"] == "scene":
+            assert emb.ndim == 1
+            nembeddings += 1
+            ndim = emb.shape[0]
+            dtype = emb.dtype
+        elif metadata["embedding_type"] == "event":
+            assert emb.ndim == 2
+            nembeddings += emb.shape[0]
+            ndim = emb.shape[1]
+            dtype = emb.dtype
+        else:
+            raise ValueError(f"Unknown embedding type: {metadata['embedding_type']}")
+
+    open(
+        embed_dir.joinpath(task_name, f"{split_name}.embedding-dimensions.json"), "wt"
+    ).write(json.dumps((nembeddings, ndim)))
+    embedding_memmap = np.memmap(
+        filename=embed_dir.joinpath(task_name, f"{split_name}.embeddings.npy"),
+        dtype=dtype,
+        mode="w+",
+        shape=(nembeddings, ndim),
+    )
+    idx = 0
+    labels = []
+    for embedding_file in tqdm(embedding_files):
+        emb = np.load(embedding_file)
+        lbl = json.load(
+            open(str(embedding_file).replace("embedding.npy", "target-labels.json"))
+        )
+        if metadata["embedding_type"] == "scene":
+            assert emb.ndim == 1
+            embedding_memmap[idx] = emb
+            # This is janky AF. The format is inconsistent with event embeddings
+            # AND its multiclass not multilabel. Honestly this should just be a list of labels.
```

Comment:
Agreed -- list of labels is great. What is inconsistent about this the event embedding format? They need to be different no? Or just that it's not a list of labels.

Reply:
I just don't think scene should be a dict and event should be a list.